src/Features/CSharp/Portable/Completion/KeywordRecommenders/FieldKeywordRecommender.cs

```diff
@@ -20,11 +22,36 @@ internal class FieldKeywordRecommender : AbstractSyntacticSingleKeywordRecommend
         SyntaxKind.EnumDeclaration,
     };
 
-    public FieldKeywordRecommender()
-        : base(SyntaxKind.FieldKeyword)
+    protected override bool IsValidContext(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
     {
+        // `[field:` is legal in an attribute within a type.
+        if (context.IsMemberAttributeContext(s_validTypeDeclarations, cancellationToken))
+            return true;
+
+        // Check if we're within a property accessor where the `field` keyword is legal.  Note: we do not do a lang
+        // version check here.  We do not want to interfere with users trying to use/learn this feature.  The user will
+        // get a clear message if they're not on the right lang version telling them about the issue, and offering to
+        // upgrade their project if they way.
+        if (context.IsAnyExpressionContext || context.IsStatementContext)
+        {
+            if (!context.IsNameOfContext && IsInPropertyAccessor(context.TargetToken))
+                return true;
+        }
+
+        return false;
     }
 
-    protected override bool IsValidContext(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
-        => context.IsMemberAttributeContext(s_validTypeDeclarations, cancellationToken);
+    private static bool IsInPropertyAccessor(SyntaxToken targetToken)
+    {
+        for (var node = targetToken.Parent; node != null; node = node.Parent)
```

Comment:
❔ Does this intentionally walk out of local functions? Even if so, it seems like static local functions should be skipped.

Reply:
yes.  it intentionally walks out of local functions.

Reply:
>  Even if so, it seems like static local functions should be skipped.

`field` is allowed ina  static local/lambda if hte containing property is also static.  We could consdier adding the logic to check for that.  but i'm in the not-caring poitn currently.